core/src/main/java/com/klaytn/caver/tx/type/TxTypeCancel.java

```diff
@@ -61,4 +65,40 @@ public static TxTypeCancel createTransaction(
         return Type.CANCEL;
     }
 
+    /**
+     * decode transaction hash from sender to reconstruct transaction with fee payer signature.
+     *
+     * @param rawTransaction signed transaction hash from sender
+     * @return TxTypeCancel decoded transaction
+     */
+    public static TxTypeCancel decodeFromRawTransaction(byte[] rawTransaction) {
+        // SenderTxHashRLP = type + encode([nonce, gasPrice, gas, from, txSignatures])
+        try {
+            byte[] rawTransactionExceptType = KlayTransactionUtils.getRawTransactionNoType(rawTransaction);
+
+            RlpList rlpList = RlpDecoder.decode(rawTransactionExceptType);
+            List<RlpType> values = ((RlpList) rlpList.getValues().get(0)).getValues();
+            BigInteger nonce = ((RlpString) values.get(0)).asPositiveBigInteger();
+            BigInteger gasPrice = ((RlpString) values.get(1)).asPositiveBigInteger();
+            BigInteger gasLimit = ((RlpString) values.get(2)).asPositiveBigInteger();
+            String from = ((RlpString) values.get(3)).asString();
+
+            TxTypeCancel tx
+                    = TxTypeCancel.createTransaction(nonce, gasPrice, gasLimit, from);
+
+            RlpList vrs = (RlpList) ((RlpList) (values.get(4))).getValues().get(0);
```

Comment:
Same here. We need to consider that we can have multiple signatures.

Reply:
Same here. This part is replaced by `protected void addSenderSignatureData (List <RlpType> signatureRlpTypeList)` after #32 merges. Now, it is a temporary implementation to suit the usability of existing caver-java users.